pkg/config/config.go

```diff
@@ -182,3 +186,49 @@ func findIsoSize(vmi *v1.VirtualMachineInstance, volume *v1.Volume, emptyIso boo
 	}
 	return 0, nil
 }
+
+type volumeInfo interface {
+	isValidType(*v1.Volume) bool
+	getSourcePath(*v1.Volume) string
+	getIsoPath(*v1.Volume) string
+	getLabel(*v1.Volume) string
+}
+
+func createIsoDisksForConfigVolumes(vmi *v1.VirtualMachineInstance, emptyIso bool, info volumeInfo) error {
+	volumes := make(map[string]v1.Volume)
+	for _, volume := range vmi.Spec.Volumes {
+		if info.isValidType(&volume) {
+			volumes[volume.Name] = volume
+		}
+	}
+
+	for _, disk := range vmi.Spec.Domain.Devices.Disks {
+		volume, ok := volumes[disk.Name]
+		if !ok {
+			log.DefaultLogger().V(2).Warningf("No matching volume with name %s found", disk.Name)
```

Comment:
I think this warning will be a bit misleading. Here, `volumes` map is populated only with volumes of a valid type `info.isValidType`. While the iteration through the disks does not take the type into account. So a disk may in fact have a matching volume, but of a different type. I would drop the warning.